pkg/dns/service.go

```diff
@@ -323,3 +324,20 @@ func (s *Service) CleanupDNSRecords(ctx context.Context, owner traffic.Interface
 	}
 	return nil
 }
+
+// getDNSRecordManagedZone returns the current ManagedZone for the given DNSRecord.
+func (s *Service) GetDNSRecordManagedZone(ctx context.Context, dnsRecord *v1alpha1.DNSRecord) (*v1alpha1.ManagedZone, error) {
+
+	if dnsRecord.Spec.ManagedZoneRef == nil {
+		return nil, fmt.Errorf("no managed zone configured for : %s", dnsRecord.Name)
+	}
+
+	managedZone := &v1alpha1.ManagedZone{}
+
+	err := s.controlClient.Get(ctx, client.ObjectKey{Namespace: dnsRecord.Namespace, Name: dnsRecord.Spec.ManagedZoneRef.Name}, managedZone)
+	if err != nil {
+		return nil, err
+	}
+
+	return managedZone, nil
+}
```

Comment:
@maleck13 is talking about removing this service, so we might want to reconsider adding this here. That said, I'm fine with leaving it and making Craig deal with it. You will need to do this in a few places, so a util function seems reasonable somewhere.

Reply:
id like to get this pr through so if things arent still concrete yet id say wait to change it when we remove the service

Reply:
agreed, lets get this merged so we can move on